Replace the remaining `Node.removeChild()` instances with `Element.remove()`

Snuffleupagus opened the pull request:
Using `Element.remove()` is a slightly more compact way of removing an element, since you no longer need to explicitly find/use its parent element.
Furthermore, the patch also replaces a couple of loops that're used to delete all elements under a node with simply overwriting the contents directly (a pattern already used throughout the viewer).

See also:
 - https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild
 - https://developer.mozilla.org/en-US/docs/Web/API/Element/remove

Snuffleupagus committed:
Replace the remaining `Node.removeChild()` instances with `Element.remove()`

Using `Element.remove()` is a slightly more compact way of removing an element, since you no longer need to explicitly find/use its parent element.
Furthermore, the patch also replaces a couple of loops that're used to delete all elements under a node with simply overwriting the contents directly (a pattern already used throughout the viewer).

See also:
 - https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild
 - https://developer.mozilla.org/en-US/docs/Web/API/Element/remove

Snuffleupagus commented:
/botio test

SamyCookie commented:
@Snuffleupagus You can activate the rule `unicorn/prefer-dom-node-remove` to reinforce that way.

cf. https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-dom-node-remove.md

Snuffleupagus committed:
Enable the `unicorn/prefer-dom-node-remove` ESLint plugin rule

Please see https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-dom-node-remove.md

timvandermeij commented:
Nice work; thanks both!
